### ISSUE TITLE
New Router Module Duplicates API Docs

### ISSUE THREAD

**Issue by mr-short** [nestjs/swagger#1451]
## I'm submitting a...
<pre><code>
[ ] Regression
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behavior
Using the new [Nestjs Router Module](https://docs.nestjs.com/recipes/router-module) creates duplicate api routes in swagger, one without the route prefix and one with it (but pushed to the end of the routes list instead of where the originals are).

e.g. I have a route `/cats` that I then use the router module to prefix `/api` with to create a `/api/cats` route. Swagger then shows:
```
/cats       <-- no longer exists since it has a prefix
/**         <-- my default catch all route that is created after all other routes
/api/cats   <-- actual route but in the wrong order
```

## Expected behavior
For swagger to correctly use the route metadata to generate only the real api routes, and in the correct position.

e.g. based on example above:
```
/api/cats
/**
```

## Minimal reproduction of the problem with instructions

Repro repo: https://github.com/mr-short/nestjs-swagger-test

- Use the new Router Module
- Nest a controller under a route prefix
- Generate swagger docs

## What is the motivation / use case for changing the behavior?

To have proper API docs

## Environment

<pre><code>
@nestjs/core@8.0.2
@nestjs/swagger@5.0.1
 
For Tooling issues:
- Node version: 14.17.1
- Platform:  Linux/Ubuntu

</code></pre>

**Comment by kamilmysliwiec** [comment]
Please provide a minimum reproduction repository.

**Comment by mr-short** [comment]
Using the stock app with simple routes the problem goes away, so I will dig into how my modules/routes are being built and see how my setup differs.

**Comment by mr-short** [comment]
@kamilmysliwiec 

It looks like it is due to the `deepScanRoutes` swagger config option being used. Is this intended?

Repro repo: https://github.com/mr-short/nestjs-swagger-test
Docs: https://docs.nestjs.com/openapi/introduction#document-options

![image](https://user-images.githubusercontent.com/6852534/125828439-6eee613f-7688-48fb-b166-eae8a1d22e6b.png)

**Comment by kamilmysliwiec** [comment]
Fixed in 5.0.4
